app/mailers/task_mailer.rb

```diff
@@ -0,0 +1,77 @@
+class TaskMailer < ApplicationMailer
+  def task_created(project, task)
+    @project_memberships = ProjectMembership.find_by(project_id: project.id)
+
+    @project_memberships.each do |member|
+      if member.role == "owner"
+        owner_create(project, task, member)
+      else
+        membership_create(project, task, member)
+      end
+    end
+  end
+
+  def task_updated(project, task)
+    @project = project
+    @task = task
+    @project_memberships = ProjectMembership.find_by(project_id: project.id)
+    
+    @project_memberships.each do |member|
+      mail(to: member.email)
+    end
+  end
+
+  def task_destroyed(project, task, current_user)
+    @project_memberships = ProjectMembership.find_by(project_id: @project.id)
+
+    @project_memberships.each do |member|
+      if member.user_id == current_user.id and member.role == "member"
```

Comment:
Please use double `&` instead of `and`, it is a bad practice

```suggestion
      if member.user_id == current_user.id && member.role == "member"
```